### PR TITLE
Fixed #29615 -- Document diff in behaviour b/w m2m_changed for add & remove, when called multiple times.

### PR DESCRIPTION
**Pull request by CuriousLearner**
@carltongibson Please let me know if you think we can improve the wording here to be more appropriate.

### CODE REVIEW

File: docs/ref/signals.txt [hunk 1]
```diff
@@ -344,6 +344,12 @@ Argument        Value
 ``using``       ``"default"`` (since the default router sends writes here)
 ==============  ============================================================
 
+__NOTE__:
+
+Multiple calls to `add()` method trigger `m2m_changed` receiver with `post_add` action only the very first time if the related object is present. All subsequent calls do not trigger `m2m_changed` due to a filtering check that happens to avoid `django.db.utils.IntegrityError`.
```
Comment (carltongibson) [inline docs/ref/signals.txt]:
```
Multiple calls to ``add()`` with the same related object will only emit a single ``m2m_changed`` signal. (All subsequent...)

In contrast, multiple calls to ``remove()``...
```

File: docs/ref/signals.txt [hunk 2]
```diff
@@ -344,6 +344,12 @@ Argument        Value
 ``using``       ``"default"`` (since the default router sends writes here)
 ==============  ============================================================
 
+__NOTE__:
```
Comment (carltongibson) [inline docs/ref/signals.txt]: Also the markup here needs to be: 

```
.. note::

    Multiple calls...
```

File: docs/ref/signals.txt (hunk 1 above)
Comment (CuriousLearner) [reply docs/ref/signals.txt]: Yeah, this sounds good to me.

File: docs/ref/signals.txt (hunk 2 above)
Comment (CuriousLearner) [reply docs/ref/signals.txt]: Oops, I'm sorry, I was traveling and jetlagged and used markdown just out of habit.

I've just made the changes.

Comment (carltongibson) [reply docs/ref/signals.txt]: No issues. :-)

**Comment by CuriousLearner** [comment]
@carltongibson All set here. Can you please take another pass?

One of the CI tests are failing, but I think the current changes are not relevant to that failure.

**Comment by timgraham** [comment]
Based on what I tested, the proposed documentation doesn't look correct. The `m2m_changed` signal is always emitted. The difference between `add()` and `remove()` is what appears in `pk_set`. I think we can use [ticket #27462](https://code.djangoproject.com/ticket/27462), and I marked the other ticket as a duplicate of that.

There might be an opportunity to add some tests in `tests/m2m_signals` if there's any doubt about the behavior.

**Comment by carltongibson** [comment]
Yes, you're right Tim. Good check, thank you!

**Comment by carltongibson** [comment]
OK, let's close this. Thanks for the input @CuriousLearner!

I've suggested #11792 for the [still open ticket](https://code.djangoproject.com/ticket/27462)

**Pull request closed without merging**